testing/citest/spinnaker_testing/spinnaker.py

```diff
@@ -367,10 +359,13 @@ def new_instance_from_bindings(cls, name, status_factory, bindings, port):
     host_platform = cls.__determine_host_platform(bindings)
     if host_platform == 'native':
       base_url = None
-      if bindings['NATIVE_HOSTNAME']:
-        host = bindings['NATIVE_HOSTNAME']
-        protocol = None if host.startswith('http://') or host.startswith('https://') else 'http://'
-        base_url = construct_base_url(host, bindings['NATIVE_PORT'] or port, protocol)
+      if bindings['NATIVE_BASE_URL']:
+        base_url = bindings['NATIVE_BASE_URL']
+      elif bindings['NATIVE_HOSTNAME']:
+        base_url = 'http://{host}:{port}'.format(
+                   host=bindings['NATIVE_HOSTNAME'],
+                   port=bindings['NATIVE_PORT'] or port)
```

Comment:
If port is a parameter, you cannot give bindings precedence and disregard the parameter.
If you dont always need port then it should be a kwarg "default_port" instead of positional.

Reply:
I believe that the port from bindings should take precedence, as the port parameter is the hardcoded default. I took the logic from the existing code: https://github.com/spinnaker/spinnaker/blob/master/testing/citest/spinnaker_testing/spinnaker.py#L364

Reply:
I agree that's what you want. I was saying that the API is misleading. You have the write policy, but the wrong syntax. At least call it a "default port" and not The Port.